functions-and-operators/group-by-modifiers.md

```diff
@@ -0,0 +1,328 @@
+TiDB does support `GROUP BY` modifiers such as `WITH ROLLUP` only under MPP mode, which means tiflash nodes are necessary. HTAP capability can supply a more flexible redistribution and regroup action for implementation of `Rollup` syntax, leading a more efficient execution flow of multidimensional aggregation.
+
+Suppose that a sales table has year, country, product, and profit columns for recording sales profitability:
+
+```sql
+CREATE TABLE sales
+(
+    year    INT,
+    country VARCHAR(20),
+    product VARCHAR(32),
+    profit  INT
+);
+
+alter table sales set tiflash replica 1;
+
+insert into sales values(2000,"China","apple",1),(2001,"Japan","banana",2),(2000,"China","lemon",3)
+```
+
+To summarize table contents per year, use a simple GROUP BY like this:
+
+```sql
+TiDB [test]> SELECT year, SUM(profit) AS profit
+                  ->        FROM sales
+                  ->        GROUP BY year;
++------+--------+
+| year | profit |
++------+--------+
+| 2001 |      2 |
+| 2000 |      4 |
++------+--------+
+2 rows in set (0.057 sec)
+```
+
+The output shows the total (aggregate) profit for each year. To also determine the total profit summed over all years, you must add up the individual values yourself or run an additional query. Or you can use ROLLUP, which provides both levels of analysis with a single query. Adding a WITH ROLLUP modifier to the GROUP BY clause causes the query to produce another (super-aggregate) row that shows the grand total over all year values:
+
+```sql
+TiDB [test]> SELECT year, SUM(profit) AS profit
+    ->        FROM sales
+    ->        GROUP BY year WITH ROLLUP;
++------+--------+
+| year | profit |
++------+--------+
+| 2000 |      4 |
+| 2001 |      2 |
+| NULL |      6 |
++------+--------+
+3 rows in set (0.029 sec)
+```
+
+The `NULL` value in the year column identifies the grand total super-aggregate line.
+
+ROLLUP has a more complex effect when there are multiple GROUP BY columns. In this case, each time there is a change in value in any but the last grouping column, the query produces an extra super-aggregate summary row.
+
+For example, without ROLLUP, a summary of the sales table based on year, country, and product might look like this, where the output indicates summary values only at the year/country/product level of analysis:
+
+```sql
+TiDB [test]> SELECT year, country, product, SUM(profit) AS profit
+    ->        FROM sales
+    ->        GROUP BY year, country, product;
++------+---------+---------+--------+
+| year | country | product | profit |
++------+---------+---------+--------+
+| 2000 | China   | lemon   |      3 |
+| 2000 | China   | apple   |      1 |
+| 2001 | Japan   | banana  |      2 |
++------+---------+---------+--------+
+3 rows in set (0.002 sec)
+```
+
+With ROLLUP added, the query produces several extra rows:
+
+```sql
+TiDB [test]> SELECT year, country, product, SUM(profit) AS profit
+    ->        FROM sales
+    ->        GROUP BY year, country, product WITH ROLLUP;
```

Comment:
Maybe add a order by clause like `order by year desc,country desc, product desc` wil make it easier to understand the following explainations?